### PR TITLE
SQL: Fix query translation for scripted queries

### PR DESCRIPTION
**Pull request by matriv**
IsNull, IsNotNull and Not where generating wrong queries
as the check to generate ScriptQuery was missing.

Fixes: #35232

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/arithmetic/Neg.java [hunk 1]
```diff
@@ -21,7 +20,7 @@
 /**
  * Negation function (@{code -x}).
  */
-public class Neg extends UnaryScalarFunction implements ScriptWeaver {
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/arithmetic/Neg.java]: ScalarFunction already implements ScriptWeaver

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java [hunk 2]
```diff
@@ -845,6 +818,14 @@ public QueryTranslation translate(Expression exp, boolean onAggs) {
 
         protected abstract QueryTranslation asQuery(E e, boolean onAggs);
 
+
+        protected static Query generateQuery(ScalarFunction sf, Expression field, Supplier<Query> query) {
```
Comment (matriv) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java]: maybe better suggestion for a name here?

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java (hunk 2 above)
Comment (costin) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java]: `wrapQuery`/`handleQuery`/`handleScript`?

**Review by costin — APPROVED** [review]
LGTM

**Comment by matriv** [comment]
Backported to `6.x` with ed9a855a789ee2c19f5ef2bd9688428db5394c1d

**Comment by matriv** [comment]
Backported to `6.5` with cab2fa1ff0b3e2d3de784ac35f5e156f3a6fe383

**Pull request merged**